app.js

```diff
@@ -238,19 +238,28 @@ const isTimeoutError = (err) => err && err.message && (
 	|| err.message.includes('ETIMEDOUT')
 );
 
-app.use((err, req, res, next) => {
+const errorHandler = (err) => {
 	const error = err.error || err;
 	const status = error.status || error.statusCode || 500;
 	error.statusCode = status;
 
-	if (!error.options) {
-		error.options = {};
+	// prevent logging jwts and x-api-keys
+	if (error.options && error.options.headers) {
+		delete error.options.headers;
+	}
+	if (error.response && error.response.request && error.response.request.headers) {
+		delete error.response.request.headers;
```

Comment:
anything in `error.options.headers` or `error.response.request.headers` that might become deleted but shouldnt be?

funny actually....why is it `error.response.request.headers` instead of `error.request.headers` ?

Reply:
true, this is unncecery / has no effect because the promise reject exception is handled by the logger outside express, in the process function
will be removed